languages/erlang/reference/implementing-a-concept-exercise.md

```diff
@@ -1,5 +1,72 @@
 # How to implement an Erlang concept exercise
 
-TODO: describe how to implement a concept exercise for the Erlang track. For inspiration, check out the [C# version of this file][csharp-implementing].
+This document describes how to implement a concept exercise for the Erlang track. As this document is generic, the following placeholders are used:
 
-[csharp-implementing]: ../../csharp/reference/implementing-a-concept-exercise.md
+- `<SLUG>`: the name of the exercise in kebab-case (e.g. `anonymous-functions`).
+- `<NAME>`: the name of the exercise in snake_case (e.g. `anonymous_functions`).
+
+Before implementing the exercise, please make sure you have a good understanding of what the exercise should be teaching (and what not). This information can be found in the exercise's GitHub issue. Having done this, please read the [Erlang concept exercises introduction][concept-exercises].
+
+To implement a concept exercise, the following files must be created:
+
+```
+languages
+└── erlang
+    └── exercises
+        └── concept
+            └── <SLUG>
+                ├── .docs
+                │   ├── instructions.md
+                │   ├── introduction.md
+                │   ├── hints.md
+                │   └── after.md (optional)
+                ├── .meta
+                │   ├── config.json
+                │   ├── design.md
+                │   └── example.erl
+                ├── <NAME>.erl
+                ├── <NAME>.hrl (optional)
+                ├── <NAME>_tests.erl (optional if <NAME>_suite.erl exists)
+                └── <NAME>_suite.erl (optional if <NAME>_tests.erl exists)
```

Comment:
This structure is different from how the existing Erlang track is structured: https://github.com/exercism/erlang/tree/master/exercises/two-fer

It is possible to move to a new structure in v3, but this would also mean converting all the existing v2 exercises to the new structure. So yes this can be changed, but it should be done for a good reason and considering the amount of work it results in.

Reply:
I would suggest using the existing structure (unless there has already been a discussion amongst the maintainers to change this). You can use the structure as defined in [this PR](https://github.com/exercism/v3/pull/213).

Reply:
This structure has 2 major flaws in my opinion:

1. It requires the student to run tests from within the erlang REPL, which already requires a certain level of understanding the language.
2. It uses 2 different testing frameworks, which according to this proposal might even alternate between individual exercises, or within one exercise co-exist as just the same set of tests using different frameworks and eventually get out of sync and in worst case scenarios contradict each other.

Therefore I used `rebar3` as build tool (just run a single easy to remember command from a regular shell) and `eunit` as testing framework, as it requires a lot less ceremony/boilerplate than `ct` does.

Also `eunit` is a unit testing tool, while `ct` is much more usefull in acceptance or integration tests.